python/taichi/lang/snode.py

```diff
@@ -12,52 +12,119 @@
 
 
 class SNode:
+    """A Python-side SNode wrapper."""
     def __init__(self, ptr):
         self.ptr = ptr
 
-    def dense(self, indices, dimensions):
+    def dense(self, axes, dimensions):
+        """Adds a dense SNode as a child component of `self`.
+
+        Args:
+            axes (List[Axis]): Axes to activate.
+            dimensions (Union[List[int], int]): Shape of each axis.
+
+        Returns:
+            The added :class:`~taichi.lang.SNode` instance.
+        """
         if isinstance(dimensions, int):
-            dimensions = [dimensions] * len(indices)
-        return SNode(self.ptr.dense(indices, dimensions))
+            dimensions = [dimensions] * len(axes)
+        return SNode(self.ptr.dense(axes, dimensions))
 
-    def pointer(self, indices, dimensions):
+    def pointer(self, axes, dimensions):
+        """Adds a pointer SNode as a child component of `self`.
+
+        Args:
+            axes (List[Axis]): Axes to activate.
+            dimensions (Union[List[int], int]): Shape of each axis.
+
+        Returns:
+            The added :class:`~taichi.lang.SNode` instance.
+        """
         if isinstance(dimensions, int):
-            dimensions = [dimensions] * len(indices)
-        return SNode(self.ptr.pointer(indices, dimensions))
+            dimensions = [dimensions] * len(axes)
+        return SNode(self.ptr.pointer(axes, dimensions))
 
-    def hash(self, indices, dimensions):
+    def hash(self, axes, dimensions):
+        """Not supported."""
```

Comment:
maybe we could add a deprecation warning later if this is not supported anymore?

Reply:
It's not exactly deprecated... Just stuck in a not supported mode when taichi switched to the LLVM backend

Reply:
I see!